src/js/core.js

```diff
@@ -421,7 +422,41 @@ const photoBooth = (function () {
                 $('#' + imgFilter).addClass('activeSidenavBtn');
 
                 if (result.error) {
-                    api.errorPic(result);
+                    if (config.picture.retry_on_error > 0 && retry < config.picture.retry_on_error) {
+                        photoboothTools.console.logDev('Taking picture failed. Retrying. Retry: ' + retry);
+                        retry += 1;
+                        loading.append(
+                            $('<p class="text-muted">').text(
+                                retrymsg + ' ' + retry + '/' + config.picture.retry_on_error
+                            )
+                        );
+                        api.startCountdown(config.picture.retry_timeout, counter, () => {
+                            loading.empty();
+                            counter.empty();
+                            cheese.empty();
+
+                            if (config.picture.no_cheese) {
+                                photoboothTools.console.log('Cheese is disabled.');
+                                api.callTakePicApi(data, retry);
```

Comment:
Pass `retry` to countdown, handle if statement to call `api.callTakePicApi` from there.